plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/CreateApplicationFromTemplateJob.java

```diff
@@ -0,0 +1,111 @@
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.job;
+
+import java.util.Collection;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Display;
+import org.jboss.tools.openshift.internal.common.core.job.AbstractDelegatingMonitorJob;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.dialog.ResourceSummaryDialog;
+import org.jboss.tools.openshift.internal.ui.util.MessageDialogUtil;
+import org.jboss.tools.openshift.internal.ui.wizard.application.IResourceLabelsPageModel.Label;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.resources.IProjectTemplateProcessing;
+import com.openshift.restclient.model.IConfig;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.template.IParameter;
+import com.openshift.restclient.model.template.ITemplate;
+
+/**
+ * A Job to create an application from a template
+ */
+public class CreateApplicationFromTemplateJob extends AbstractDelegatingMonitorJob {
+
+	private IProject project;
+	private ITemplate template;
+	private Collection<Label> labels;
+	private Collection<IParameter> parameters;
+
+	public CreateApplicationFromTemplateJob(IProject project, ITemplate template, Collection<IParameter> parameters, Collection<Label> labels) {
+		super("Create Application From Template");
+		this.project = project;
+		this.template = template;
+		this.labels = labels;
+		this.parameters = parameters;
+	}
+
+	@Override
+	protected IStatus doRun(IProgressMonitor monitor) {
+		template.updateParameterValues(parameters);
+		for (Label label : labels) {
+			template.addLabel(label.getName(), label.getValue());
+		}
+		
+		IStatus status = project.accept(new CapabilityVisitor<IProjectTemplateProcessing, IStatus>() {
+
+			@Override
+			public IStatus visit(IProjectTemplateProcessing capability) {
+				
+				try {
+					IConfig config = capability.process(template);
+					
+					Collection<IResource> resources = capability.apply(config);
+					return handleResponse(resources);
+				}catch(OpenShiftException e) {
+					String message = e.getMessage();
+					if(e.getStatus() != null) {
+						message = e.getStatus().getMessage();
+					}
+					showErrorDialog(message);
```

Comment:
manually opening a dialog from a job usually is not required. The job runner/pool in the platform is showing a dialog all auto-magically if you return a correctly setup up error-status.